src/lib/logstore/log_dev.cpp

```diff
@@ -512,10 +514,16 @@ void LogDev::on_flush_completion(LogGroup* lg) {
     auto upto_indx = lg->m_flush_log_idx_upto;
     auto dev_offset = lg->m_log_dev_offset;
     for (auto idx = from_indx; idx <= upto_indx; ++idx) {
-        auto& record = m_log_records->at(idx);
-        logstore_req* req = s_cast< logstore_req* >(record.context);
+        logstore_req* req;
+        logstore_id_t store_id;
+        {
+            folly::SharedMutexWritePriority::WriteHolder holder(m_stream_tracker_mtx);
```

Comment:
Can we put a comment to describe the race here? It would help us understanding the reason for this lock we are holding here.